pkg/util/exec/exec.go

```diff
@@ -271,15 +252,19 @@ func runCommandWithOutput(actionName string, cmd *exec.Cmd, combinedOutput bool)
 	out := strings.TrimSpace(string(output))
 
 	if err != nil {
-		return out, createCommandError(err, actionName)
+		return out, err
 	}
 
 	return out, nil
 }
 
-func logCommand(debug bool, command string, arg ...string) {
+func logCommand(suppressLogOutput bool, command string, arg ...string) {
 	msg := fmt.Sprintf("Running command: %s %s", command, strings.Join(arg, " "))
-	if debug {
+	writeToLog(suppressLogOutput, msg)
+}
+
+func writeToLog(suppressLogOutput bool, msg string) {
+	if suppressLogOutput {
```

Comment:
Why don't we align this based on the log level of the operator? That could simply the arguments, especially for `ExecuteCommandWithOutput`. Currently, it's mostly `true` if we have properly log ingo on commands we are executing.

Reply:
Ok, I still think we need better control but I can see that some packages will always run DEBUG, such as:

* `pkg/daemon/ceph/osd/volume.go`
* `pkg/util/sys/kmod.go`
* `pkg/util/sys/device.go`

So perhaps we should just always turn on DEBUG since these packages are not that verbose anyway.

We should concentrate the usefulness of the logs on the operator logs IMHO.

Reply:
Yes, the point of this change was to simplify the logs from the point of view of the operator. At the code level, it's up to the dev to decide whether their commands should be logged at Info or Debug level. At runtime, it's up to the cluster admin to decide if Info or Debug is enabled. We might be saying the same thing. Let's discuss to clarify.

Reply:
Per discussion, to improve clarity I'll make the following changes:
- The exec package will always log the command and output at debug level. Then we can remove the confusing `debug` param from all the exec methods.
- Enable debug logging in the osd prepare pod since all the commands and output are interesting in that job.